std/net/curl.d

```diff
@@ -1069,12 +1121,23 @@ private auto _basicHTTP(T)(const(char)[] url, const(void)[] sendData, HTTP clien
             import std.conv : to;
             content.reserve(value.to!size_t);
         }
+        header ~= cast(ubyte[])(key ~ ": " ~ value ~ "\n");
     };
     client.onReceiveStatusLine = (HTTP.StatusLine l) { statusLine = l; };
     client.perform();
     enforce(statusLine.code / 100 == 2, new HTTPStatusException(statusLine.code,
             format("HTTP request returned status code %d (%s)", statusLine.code, statusLine.reason)));
 
+    if (client.method == HTTP.Method.head)
+    {
+        import std.conv : to;
+
+        auto tmpHeader = _decodeContent!T(header.data, client.p.charset);
+        auto statusFormat = to!(typeof(tmpHeader))(format("HTTP/%s.%s %s %s\n", statusLine.majorVersion,
+                statusLine.minorVersion, statusLine.code, statusLine.reason));
```

Comment:
Oh no, so this doesn't actually even return the headers / status line as it was sent by the HTTP server - it just builds a string that looks like one!

Sorry, I can't imagine there being a justifiable use case for this.